### PR TITLE
[READY] Allow creation of the various types of emergency oxygen tanks

### PR DESCRIPTION
**Pull request by ghost**
[Changelogs]:

:cl: HideAndSeekLOGIC
add: Added emergency oxygen tanks and extended capacity emergency tanks to the autolathe and protolathe
add: Added them to the Industrial Engineering tech node
tweak: Only Cargo and Engineering can print extended capacities; the rest are available for everyone.
/:cl:

[why]: # 

There is currently no way to produce these tanks and there is currently no reason why this should be the case. This PR fixes this problem.

### CODE REVIEW

**Comment by k-k-karma** [comment]
Verified Webeditor

**Comment by Barhandar** [comment]
Significantly reduces desirability of advanced pocket oxytanks. Also, double is higher capacity than engineering. Also, POCKET BOMBS FOR EVERYONE.

**Comment by Steelpoint** [comment]
I really, really, disagree with making double capacity emergency oxygen tanks something that people can print out. 

It used to be something that only the Deathsquad would get, not even the Emergency Response Team gets them. Then Deltastation ruined it by spawning em on the arrivals ship.

I could maybe get behind making it some rare item for the CE or akin to have, but I'd rather keep it as a Deathsquad/Centcom only item.

**Comment by Tlaltecuhtli** [comment]
make the double air tanks require alien tech

**Comment by anconfuzedrock** [comment]
I Was going to write 20 paragraphs but steelpoint already said it

Not even EMERGENCY RESPONSE TEAMS get extended capacities. it's really just deltastation powercreep.

**Comment by HideAndSeekLOGIC** [comment]
@Steelpoint from what I can remember, miners start with a double capacity as well, so there goes that argument. 

In addition, saying 'but only Deathsquad gets them' is just fucking dumb - a tank is nowhere near as strong as any of the other ERT/DS items. Another factor here is that these particular tanks come out empty - you have to fill them up yourself, unlike DS.

The entire point of this PR was to reward those who bother printing out these tanks, creating an air pump with pure oxygen and using it to fill the tank with a highly versatile item containing an extended duration of air - this is still highly situational and is only useful in extremely niche cases.

@anconfuzedrock again, dumb argument. ERT doesn't get a great deal of things, should we remove all those items from the game as well? Also, if you think having slightly longer air capacity is powercreep then you're most definitely wrong.

@Tlaltecuhtli it's literally just a bigger tank. I'd put it behind illegal tech at most, but I still don't think it's anywhere near as strong as to be put behind any such nodes.

**Comment by anconfuzedrock** [comment]
@HideAndSeekLOGIC 
Miners fucking don't, so there goes that argument
In addition, saying 'but only deathsquad gets them' is fucking fair- it's not as strong as the rest of their kit, but it's a signature item of the deathsquad. When double caps were added, they were added as an item to make deathsquad life easier, while recognising that it isn't fair to give the station such easy air.  you have to fill em yourself, sure, but that's incredibly easy. if anything, all small oxygen tanks need nerfed. Ert doesn't get too great a deal of things, but they get powerful pulse weaponry at their height, and at that height they still don't get double caps. And yes, it's powercreep. it's a slight buff that isn't needed, that's the definition of powercreep.

**Comment by anconfuzedrock** [comment]
If the tanks start empty, an extended capacity tank that's completely filled with oxygen is going to last you for an infinitely long time anyways since air tanks hold so much already. you don't need to give them double size tanks, it's never going to matter, so just don't.

**Comment by HideAndSeekLOGIC** [comment]
@anconfuzedrock @Steelpoint Right, I was wrong about the miners and thus I apologise. If you all are so against including the double tanks (which I still don't really understand the motivation for), I'll remove them from this PR.

**Comment by erwgd** [comment]
I'd also add plasmaman internals tanks, specifically /obj/item/tank/internals/plasmaman/belt. Since these have the same volume as engineering tanks they should be similarly exclusive to engineering.

**Comment by HideAndSeekLOGIC** [comment]
@erwgd Yep, I'll also gave them to cargo though.

**Comment by alexkar598** [comment]
This isn’t going to affect anything other than tank color. Since they start empty,people need to fill it at an o2 tank but they can’t put more moles than in other tanks due to the tank KPA limit meaning atmos ans maybe engineers are the only ones who will be able to fit more moles in the same amount of KPAs.

**Comment by Jalleo** [comment]
@HideandseekLogic I downvoted your PRbecause I believe thisis pure feature creep for feature creep.


Outside of having lots of spare tanks stored in your box with oxygen what use is there to having this?

Internals should be rarer to make this a worthwhile thing to add.

**Comment by HideAndSeekLOGIC** [comment]
@Jalleo This has a use and I already explained it. This PR exists to reward those who took the time to prepare for a threat with the tools to avoid that threat, whether that threat be depressurised rooms, AI flooding or some greytide running around with N2O. Currently, only science and engineering have ways around that, and it's not the most convenient way to go about things. Additionally, only large oxygen tanks can be created at the time, which generally have enough air to last a full 2 hour round (albeit with the issue of having to carry it around in a hand or backpack slot).

Allowing all departments and autolathes to create emergency tanks changes this. If a person takes the time and effort to fill up these, they are then rewarded with the ability to pass through most air hazards unharmed for extended periods of time, generally regardless of their job. Also, it lets people who lost their original tank still have an emergency air supply.

Do note that tanks only give the ability to breathe clean air and nothing else. Vacuum, hot plasma, low pressure, etc. is still a massive threat for the player, even if they do indeed have a box filled with these.

Thanks for taking your time to explain the reasoning behind your downvote, I truly appreciate it.

**Comment by Jalleo** [comment]
That is exactly the reason why it should be limited. It is better to be a slightly scarce resource. It is a way to force people to not have internals on all the time to avoid viruses and other pressure or air based hazards.
Also the emergency service oxygen tanks are not only limited to the one in your box oxygen lockers can have then too.

Lastly you said about convenience yes it would be convenient to have this. The issue though is you effectively create a way to powergame to totally avoid this issue by mass production. Ordering of internals crates are a better way to manage this.

This PR only serves to make it easier to avoid problems which we have available for the antagonists to cause. Sadly not all the time is it fun or enjoyable but if we prevent all these 'unfortunate' accidents by creating ways to avoid eventually it might become almost impossible to be a antagonist without massive skill and luck. Due to one aspect people being able to avoid the issues caused by ease of availability.

**Comment by anconfuzedrock** [comment]
Oh, I just hadn’t looked at the pr in a bit. That simple.

**Comment by ExcessiveUseOfCobblestone** [comment]
This seems to go against some of the other PRs in which internals are being limited/stifled to make it more valuable.

File: code/game/objects/items/tanks/tank_types.dm [hunk 1]
```diff
@@ -154,6 +154,11 @@
 	..() // Plasma asserted in parent
 	air_contents.gases[/datum/gas/plasma][MOLES] = (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
 	return
+	
+/obj/item/tank/internals/plasmaman/belt/empty/New()
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]:
```suggestion
/obj/item/tank/internals/plasmaman/belt/empty/Initialize()
```

File: code/game/objects/items/tanks/tank_types.dm [hunk 2]
```diff
@@ -154,6 +154,11 @@
 	..() // Plasma asserted in parent
 	air_contents.gases[/datum/gas/plasma][MOLES] = (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
 	return
+	
+/obj/item/tank/internals/plasmaman/belt/empty/New()
+	..() // Plasma asserted in parent
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]:
```suggestion
	. = ..() // Plasma asserted in parent
```

File: code/game/objects/items/tanks/tank_types.dm [hunk 3]
```diff
@@ -154,6 +154,11 @@
 	..() // Plasma asserted in parent
 	air_contents.gases[/datum/gas/plasma][MOLES] = (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
 	return
+	
+/obj/item/tank/internals/plasmaman/belt/empty/New()
+	..() // Plasma asserted in parent
+	air_contents.gases[/datum/gas/plasma][MOLES] = 0
+	return
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]:
```suggestion
```

File: code/game/objects/items/tanks/tank_types.dm [hunk 4]
```diff
@@ -177,13 +182,29 @@
 	air_contents.assert_gas(/datum/gas/oxygen)
 	air_contents.gases[/datum/gas/oxygen][MOLES] = (3*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
 	return
+	
+/obj/item/tank/internals/emergency_oxygen/empty/New()
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]:
```suggestion
/obj/item/tank/internals/emergency_oxygen/empty/Initialize()
```

File: code/game/objects/items/tanks/tank_types.dm [hunk 5]
```diff
@@ -177,13 +182,29 @@
 	air_contents.assert_gas(/datum/gas/oxygen)
 	air_contents.gases[/datum/gas/oxygen][MOLES] = (3*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
 	return
+	
+/obj/item/tank/internals/emergency_oxygen/empty/New()
+	..()
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]:
```suggestion
	. = ..()
```

File: code/game/objects/items/tanks/tank_types.dm [hunk 6]
```diff
@@ -177,13 +182,29 @@
 	air_contents.assert_gas(/datum/gas/oxygen)
 	air_contents.gases[/datum/gas/oxygen][MOLES] = (3*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
 	return
+	
+/obj/item/tank/internals/emergency_oxygen/empty/New()
+	..()
+	air_contents.gases[/datum/gas/oxygen][MOLES] = 0
+	return
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]:
```suggestion
```

File: code/game/objects/items/tanks/tank_types.dm [hunk 7]
```diff
@@ -177,13 +182,29 @@
...
 
 /obj/item/tank/internals/emergency_oxygen/engi
 	name = "extended-capacity emergency oxygen tank"
 	icon_state = "emergency_engi"
 	volume = 6
 
+/obj/item/tank/internals/emergency_oxygen/engi/empty/New()
+	..()
+	air_contents.gases[/datum/gas/oxygen][MOLES] = 0
+	return
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]: Same

File: code/game/objects/items/tanks/tank_types.dm [hunk 8]
```diff
@@ -177,13 +182,29 @@
...
+
 /obj/item/tank/internals/emergency_oxygen/double
 	name = "double emergency oxygen tank"
 	icon_state = "emergency_double"
 	volume = 10
+	
+/obj/item/tank/internals/emergency_oxygen/double/empty/New()
+	..()
+	air_contents.gases[/datum/gas/oxygen][MOLES] = 0
+	return
```
Comment (Cyberboss) [inline code/game/objects/items/tanks/tank_types.dm]: Same

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/tanks/tank_types.dm (hunk 8 above)
Comment (HideAndSeekLOGIC) [reply code/game/objects/items/tanks/tank_types.dm]: Your code seems to not pass the tests. Do you know why?

**Comment by HideAndSeekLOGIC** [comment]
@ExcessiveUseOfCobblestone I'd say being able to print empty tanks in an autolathe strikes a nice balance between restricting internals and giving more internals options. These do require an air pump or canister to fill up (which most probably won't be able to get under normal circumstances) so someone can't just break into cargo and grab 30 tanks. If they're willing to then break into engineering or science to fill them up, power to them I guess. It'd be the equivalent of getting welding shield eyes - a lot of effort and preparation for something that doesn't happen often.

Also, this allows for the station to more quickly recuperate after bombings or similar mass breaches - a well organised cargo, engineering or science team would now be able to provide portable air tanks to a good portion of the crew, instead of having most die to the oxygen loss in 10 minutes.

**Comment by HideAndSeekLOGIC** [comment]
@Cyberboss I fixed the code, is there anything else I need to do?

**Comment by optimumtact** [comment]
Is the double capacity tank removed? if not, remove it

**Comment by HideAndSeekLOGIC** [comment]
@optimumtact not from the game, no. Although, it does remain a deathsquad-only item if that was your question.

**Pull request merged**